kore/src/Kore/Step/Simplification/Equals.hs

```diff
@@ -181,18 +182,21 @@ simplifyEvaluated predicate first second
   | otherwise = do
     let isFunctionConditional Conditional {term} = isFunctionPattern term
     case (firstPatterns, secondPatterns) of
-        ([firstP], [secondP]) -> makeEvaluate firstP secondP predicate
+        ([firstP], [secondP]) ->
+            makeEvaluate firstP secondP predicate
         ([firstP], _)
             | isFunctionConditional firstP ->
                 makeEvaluateFunctionalOr predicate firstP secondPatterns
         (_, [secondP])
             | isFunctionConditional secondP ->
                 makeEvaluateFunctionalOr predicate secondP firstPatterns
-        _ ->
-            makeEvaluate
-                (OrPattern.toPattern first)
-                (OrPattern.toPattern second)
-                predicate
+        _ -> if OrPattern.isPredicate first && OrPattern.isPredicate second
```

Comment:
I think a guard here would be more consistent with the cases above than using `if`, but I don't feel strongly.

Reply:
Fixed.